nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,818 @@
+from .._utils.compat import _basestring
+from .._utils.cache_mixin import _check_memory
+from .._utils.param_validation import check_feature_screening
+from .._utils import CacheMixin
+
+
+SUPPORTED_ESTIMATORS = dict(
+    svc_l1=LinearSVC(penalty='l1', dual=False),
+    svc_l2=LinearSVC(penalty='l2'),
+    svc=LinearSVC(penalty='l2'),
+    logistic_l1=LogisticRegression(penalty='l1', solver='liblinear'),
+    logistic_l2=LogisticRegression(penalty='l2', solver='liblinear'),
+    logistic=LogisticRegression(penalty='l2'),
+    ridge_classifier=RidgeClassifier(),
+    ridge_regression=Ridge(),
+    ridge=Ridge(),
+    svr=SVR(kernel='linear'),
+)
+
+
+def _check_param_grid(estimator, X, y, param_grid):
+    """Check param_grid and return sensible default if none is given.
+    For more information, see the documentation:
+    <http://http://scikit-learn.org/stable/modules/generated/sklearn.svm.l1_min_c>
+    """
+    if param_grid is None:
+        param_grid = {}
+        # define loss function
+        if isinstance(estimator, LogisticRegression):
+            loss = 'log'
+        elif isinstance(estimator, (LinearSVC, _BaseRidge, SVR)):
+            loss = 'squared_hinge'
+        else:
+            raise ValueError("%s is an unknown estimator."
+                             "The supported estimators are: %s" %
+                             (estimator, list(SUPPORTED_ESTIMATORS.keys())))
+
+        # loss = l2 was deprecated after version 0.17
+        if LooseVersion(sklearn.__version__) <= LooseVersion('0.17'):
+            if loss == 'squared_hinge':
+                loss = 'l2'
+
+        if hasattr(estimator, 'penalty') and (estimator.penalty == 'l1'):
+            min_c = l1_min_c(X, y, loss=loss)
+        else:
+            min_c = 0.5
+
+        if not isinstance(estimator, _BaseRidge):
+            param_grid['C'] = np.array([2, 20, 200]) * min_c
+        else:
+            param_grid['alpha'] = 1. / np.array([1, 10, 100])
+
+    return param_grid
+
+
+def _parallel_fit(estimator, X, y, train, test, param_grid, is_classif, scorer,
+                  mask_img, class_index, screening_percentile=None):
+    """Find the best estimator for a fold within a job.
+    This function performs the fitting of several estimators, saving the best
+    performing ones per cross-validation fold. These models are used afterwards
+    to build the averaged model.
+    """
+    n_features = X.shape[1]
+    # Checking if the size of the mask image allow us to perform feature
+    # screening
+    selector = check_feature_screening(screening_percentile, mask_img,
+                                       is_classif)
+    do_screening = (n_features > 100) and (screening_percentile < 100.)
+
+    X_train = X[train].copy()
+    y_train = y[train].copy()
+    X_test = X[test].copy()
+    y_test = y[test].copy()
+
+    if (selector is not None) and do_screening:
+        X_train = selector.fit_transform(X[train], y[train])
```

Comment:
It's weird to apply selector on X[train] and y[train] after copying them on the line above. I guess it should be applied on the copies X_train, y_train and below X_test